### PR TITLE
1957 embed pretty

### LINKED ISSUE #1957: Embedding flourishes for the gradio web component
We used to have a nice gray border around embedded gradio demos with some additional information like this at the bottom:

image

I think this would a nice enhancement to bring back to embedded gradio web components and would distinguish the web component from an ordinary iframe embedding.

TLDR: when embedding a Gradio app from Spaces, it would be nice to have links to the original Space and code.

### LINKED ISSUE #2432: Better messaging/UI for a broken embedded demo
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
Currently if an embedded gradio demo is broken it will just not show up at all on the page. Which is a bit jarring and confusing for users of the page. for example: 

Screen Shot 2022-10-11 at 2 32 39 PM

**Describe the solution you'd like**  
When the demo is broken or can't embed, etc, it should display some sort of message indicating that. This can easily be done for the website but should be done on the library side so that it works everywhere gradio demos are embedded.

### PR DESCRIPTION
**Pull request by pngwn**
# Description

This PR improves the web component embedding experience in a number of ways:

**Embed chrome**

- Embedded gradio apps now display a border and an info panel at the bottom of the embed linking back to the original space, linking to gradio and linking to spaces:
Screenshot 2023-02-17 at 12 26 12

- The info panel can be disabled by passing `info="false"`
Screenshot 2023-02-17 at 12 28 04

- The embed container can be completely disabled by passing `container="false"`:
Screenshot 2023-02-17 at 12 27 12

- To avoid the embedding behaviour entirely (for 'full' gradio apps with nice margins and things) then `embed="false"` must be passed to the web container.

**Space 'stage' handling**

This PR also introduces more graceful handling of the various spaces statuses. a Space can be:
- `STOPPED` ([soon `SLEEPING`](https://github.com/huggingface/spaces-app-manager/issues/568), this PR already handles this),
- `BUILDING` 
- `RUNNING` (or `RUNNING_BUILDING` which is running but with a build going on in the background), 
- One of several error states `NO_APP_FILE`, `CONFIG_ERROR`, `BUILD_ERROR`,  `RUNTIME_ERROR`. 

This status is now displayed when loading the space. Sleeping spaces will be woken up. Of particularly note are the error states which have been combined into one (mostly) for the end user but have slightly different behaviour. This is what they will see if the space exists and discussions are enabled:

Screenshot 2023-02-17 at 12 38 03

Clicking on the link provided will take the user to the discussion for that space with a prefilled out title and body. Including the actual error type the space is experiencing.

https://user-images.githubusercontent.com/12937446/219656329-49074968-f94f-4aa5-863e-9611c7d6e496.mov

If discussions are not enabled of the space cannot be found, then a generic error message is presented to the user and they are encouraged contact the website author instead.

**Performance improvements**

The embed chrome, some css + fonts, the space meta info and the gradio config are the only things loaded up front. Everything else is lazy loaded and rendered only when the embed is actually on the screen. This significantly improves performance for pages with a number of demos such as long blog posts or documentation sites.

With around 10 or so moderately complex gradio demos, the lighthouse score of a simple page was decreased from ~100 to ~60. With this PR that impact is negligible. See @aliabd's comment for an example (https://github.com/gradio-app/gradio/pull/3205#pullrequestreview-1303016276).

This lazy loading can be disabled by passing `eager="true"` to the custom element.

Closes #1957. Closes #2432.

### CODE REVIEW

File: demo/file_component/run.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
-import gradio as gr 
+import gradio as gr
 
-css = "footer {display: none !important;} .gradio-container {min-height: 0px !important;}"
 
-with gr.Blocks(css=css) as demo:
+with gr.Blocks() as demo:
     gr.File()
```
Comment (abidlabs) [inline demo/file_component/run.py:6]: How come we only removed the `css` from this component, wheareas all of the other `*_component` demos still include this css?

Comment (pngwn) [reply demo/file_component/run.py:6]: That was a leftover, will revert. We can remove some of these custom things now but I'll do that in a follow-up to reduce noise.

**Review by freddyaboulton — COMMENTED** [review]
This is awesome @pngwn ! Works really well.

Which version should I test with? I tested with `https://gradio.s3-us-west-2.amazonaws.com/3.18.1b5/gradio.js` and everything is working great but I don't see the cool error statuses in the embedded demos you showcased.

I just see the status tracker spinning forever


Tested with this document

```
<script type="module"
src="https://gradio.s3-us-west-2.amazonaws.com/3.18.1b5/gradio.js">
</script>



<div>
<gradio-app space="gradio/native_plots"></gradio-app>
</div>

<div>
    <gradio-app space="freddyaboulton/blocks_random_slider"></gradio-app>
</div>


<div>
    <gradio-app space="gradio/cancel_events2"></gradio-app>
</div>

<div>
    <gradio-app space="freddyaboulton/Chat-GPT-LangChain
    "></gradio-app>
</div>


<div>
    <gradio-app space="gradio/video_component"></gradio-app>
</div>

<div>
    <gradio-app space="gradio/main_note"></gradio-app>
</div>

<div>
    <gradio-app space="gradio/model3D"></gradio-app>
</div>
```

This demo is embedded in the page https://huggingface.co/spaces/freddyaboulton/Chat-GPT-LangChain and has a Runtime error so I was expecting to see something in the embedded page.

**Comment by pngwn** [comment]
Thanks @freddyaboulton, I'll take a look!

**Comment by pngwn** [comment]
@freddyaboulton Can you try again with `https://gradio.s3-us-west-2.amazonaws.com/3.18.1b6/gradio.js` and let me knwo if that fixes the issues you were seeing.

**Comment by pngwn** [comment]
Thanks for the review!

I tried with the logos previously but @gary149 and @abidlabs felt it was a little too intrusive, so we went with just the text. I used the hugging face emoji to avoid write the text (which is very long and causes issues on mobile). 

Those lighthouse scores are awesome! Very satisfying. 

You can avoid the wrapper by passing 'minimal="true"' to the web component (need to test this again but should work). There isn't currently a way to keep the border but remove the text. This might be something we want to support (or make that the behaviour by default), so I'll check.

**Review by freddyaboulton — APPROVED** [review]
@pngwn Thanks for the fix!

It works well with the latest version you published! The one thing is that the "let the author know" link is broken for me.

Good to merge once that's fixed in my opinion.

create_discussion_link_broken

**Pull request merged**